dotcom-rendering/src/web/components/Card/Card.tsx

```diff
@@ -277,6 +277,11 @@ export const Card = ({
 		return <Snap snapData={snapData} />;
 	}
 
+	// Decide what type of image to show, main media, avatar or none
+	let imageType: 'mainmedia' | 'avatar' | undefined;
```

Comment:
This variable is used to determine when an avatar should show. By using a variable like this we ensure we never show _both_ the main media image and the avatar at the same time

Reply:
I think the best approach for stuff like this is to extract it to a helper function, rather that mutating values inline.

```ts
const getImage = ({ imageUrl, avatarUrl }): ['avatar' | 'mainmedia', string] | [undefined, undefined] => {
  if(avatarUrl) return ['avatar', avatarUrl];
  if(imageUrl) return ['mainmedia', imageUrl]:
  return [undefined, undefined];
}

// inside the component:
const [imageType, image] = getImage({ imageUrl, avatarUrl });
```

Reply:
I don’t think the curly braces are much different.

The benefit of the abstraction is that you can lean more heavily on the compiler, which will know that if `imageType` is set to `"avatar"`, an `avatarUrl` will be defined. It seems to not be able to infer that from the if logic.

Reply:
If we only ever show one image, that is either the avatar or the image, could we discriminate the type for this?

Reply:
I tried creating an abstraction here but I am still getting typescript errors 😐 

```ts
	const decideGraphic = (imageUrl?: string, avatarUrl?: string) => {
		if (imageUrl && avatarUrl) {
			return 'avatar';
		} else if (imageUrl) {
			return 'image';
		}
		return 'none';
	};
```

I think the only way to avoid these would be to create a _new_ variable for the `imageUrl` and `avatarUrl`, like `theImageSourceToUse`. But adding extra complexity and abstraction to rename two variables to one value just to avoid a guard in the code doesn't seem like a good approach to me.

Reply:
Fair enough.